File: DSA.SPA/ClientApp/pages/facility/api/facility-service.js
import * as httpService from '../../../api/http-service';

const apiFacility = "/api/facility/";

export const getFacilitiesByTerm = term => {
    let params = {
        url: apiFacility + `GetFacilitiesByTerm?term=${term}`
    }

    return httpService.getData(params);
}

export const getFacilities = data => {
    let params = {
        url: apiFacility + "getFacilities",
        data: data
    }

    return httpService.postData(params);
}

export const addFacility = data => {
    let params = {
        url: apiFacility + "addFacility",
        data: data
    }

    return httpService.postData(params);
}

File: DSA.BLL/Services/FacilityService.cs
using System.Collections.Generic;
using RCS.BLL.Dto.Facilities;
using RCS.BLL.Services.Contracts;
using RCS.DAL.UnitOfWork.Contract;
using RCS.Domain.Facilities;
using RCS.Domain.Params;

namespace RCS.BLL.Services
{
    public class FacilityService : IFacilityService
    {
        private readonly IUnitOfWork _unitOfWork;

        public FacilityService(IUnitOfWork unitOfWork)
        {
            _unitOfWork = unitOfWork;
        }
        
        public void AddFacility(AddFacilityDto data)
        {
            var newFacility = AutoMapper.Mapper.Map<AddFacilityDto, Facility>(data);
            _unitOfWork.FacilityRepository.Add(newFacility);
            _unitOfWork.Commit();
        }
        
        public CollectionResult<FacilityDto> GetFacilitiesByParams(FacilitiesFilterParams filterParams)
        {
            var facilities = _unitOfWork.FacilityRepository.GetFacilitiesByParams(filterParams);

            var result = new CollectionResult<FacilityDto>
            {
                TotalCount = facilities.TotalCount,
                Collection = AutoMapper.Mapper.Map<IEnumerable<Facility>, List<FacilityDto>>(facilities.Collection)
            };

            return result;
        }

        public IEnumerable<FacilityDto> GetFacilitiesByTerm(string term)
        {
            var facilities = _unitOfWork.FacilityRepository.GetFacilitiesByTerm(term);
            return AutoMapper.Mapper.Map<IEnumerable<Facility>, List<FacilityDto>>(facilities);
        }
    }
}
